pkg/segment/reader/segread/multicolreader.go

```diff
@@ -189,9 +189,15 @@ func InitSharedMultiColumnReaders(segKey string, colNames map[string]bool, block
 		fName := fName
 		currFd, err := os.OpenFile(fName, os.O_RDONLY, 0644)
 		if err != nil {
-			log.Errorf("qid=%d, InitSharedMultiColumnReaders: failed to open file %s for columns %s. Error: %v.",
-				qid, fName, colName, err)
-			continue
+			// This segment may have been recently rotated; try reading the
+			// rotated segment file.
+			rotatedFName := writer.GetRotatedVersion(fName)
+			currFd, err = os.OpenFile(rotatedFName, os.O_RDONLY, 0644)
+			if err != nil {
+				log.Errorf("qid=%d, InitSharedMultiColumnReaders: failed to open file %s for columns %s. Error: %v.",
```

Comment:
in case if the original err was not a active/final issue, do we want to also log that err here and maybe add some a couple of words something of : `orig fname: %v had failed with err: %v, rotated also failed with fname: %v, newErr: %v`